pkg/parser/parser.go

```diff
@@ -80,6 +84,14 @@ func New(input string, builtins map[string]*FuncDecl) *Parser {
 		p.advanceTo(i)
 		fd := p.parseFuncDeclSignature()
 		if fd != nil {
+			if p.funcs[fd.Name] != nil {
```

Comment:
perhaps a switch?:
```
switch {
case p.funcs[fd.Name] == nil:
  p.funcs[fd.Name] = fd
case builtins[fd.Name] == nil:
  p.appendErrorForToken("redeclaration of function "+fd.Name, fd.Token)
case builtins[fd.Name] != nil: // could be `default:`, but maybe this reads better?
  p.appendErrorForToken("cannot override builtin function "+fd.Name, fd.Token)
}
```

Reply:
or perhaps swap the last two cases?

Reply:
ok.

Reply:
```go
switch {
case  p.funcs[fd.Name] != nil && builtins[fd.Name] == nil:
  p.appendErrorForToken("redeclaration of function "+fd.Name, fd.Token)
case p.funcs[fd.Name] != nil && builtins[fd.Name] != nil: 
  p.appendErrorForToken("cannot override builtin function "+fd.Name, fd.Token)
default:
  p.funcs[fd.Name] = fd
}
```

Reply:
any of them. i'll got with your suggestion.

Reply:
I deliberately didn't do it your way since you can take advantage of the case ordering to put the happy path first and not require the more complicated case expressions. but up to you

Reply:
but we don't do the happy path first typically.
we do it left and last, don't we?

Reply:
but rules are made to be broken. happy path is more about less nesting than what comes first. it just ends up putting the happy path last mostly.